src/main/java/frc/robot/vision/VisionDriveSubsystem.java

```diff
@@ -0,0 +1,47 @@
+package frc.robot.vision;
+
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.geometry.Translation2d;
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj.shuffleboard.Shuffleboard;
+import edu.wpi.first.wpilibj.shuffleboard.ShuffleboardTab;
+import frc.robot.Robot;
+import frc.robot.drive.DriveSubsystem;
+import frc.robot.poseScheduler.PoseScheduler;
+
+public class VisionDriveSubsystem extends DriveSubsystem {
+
+  private final IVision impl;
+  private final PoseScheduler poseScheduler;
+
+  public static VisionDriveSubsystem create(PoseScheduler poseScheduler) {
+    return new VisionDriveSubsystem(Robot.isReal() ? new RealVision() : new SimVision(), poseScheduler);
+  }
+    
+  public VisionDriveSubsystem(IVision implementation, PoseScheduler poseScheduler){
+    impl = implementation;
+    this.poseScheduler = poseScheduler;
+
+    ShuffleboardTab tab = Shuffleboard.getTab("Vision");
+    tab.addBoolean("Found target: ", () -> impl.hasTarget());
+    tab.addInteger("Target Id: ", () -> impl.getTargetId());
+    tab.addString("BotPose: ", () -> getVisionBotpose().toString());
```

Comment:
I don't think we want to log this as a string, find an example of someone logging their bot pose, or we might just be able to advantagescope the table coming off the limelight. But I don't think the string will do any good regardless